crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -752,14 +752,15 @@ impl CumulocityConverter {
                     continue;
                 }
             };
-            let child_topic_id =
-                EntityTopicId::default_child_device(child_external_id.as_ref()).unwrap();
+
+            let child_name = self.default_device_name_from_external_id(&child_external_id);
+            let child_topic_id = EntityTopicId::default_child_device(&child_name).unwrap();
```

Comment:
This `unwrap()` has to be removed as a maliciously crafted device name will crash the mapper.

Reply:
Not urgent. The error is catched before.

```
Supported operations of child device directory: demo-device-777:device:+ ignored due to Invalid external ID: demo-device-777:device:+ contains invalid character: +    
```

Reply:
Fixed